pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -145,16 +146,16 @@ func (c *APIClient) connect() error {
 	return nil
 }
 
-// MetadataMapperBundle maps pod names to associated metadata.
-type MetadataMapperBundle struct {
-	Services ServicesMapper `json:"services,omitempty"`
-	mapOnIP  bool           // temporary opt-out of the new mapping logic
+// metadataMapperBundle maps pod names to associated metadata.
+type metadataMapperBundle struct {
+	Services apiv1.NamespacesPodsStringsSet
+	mapOnIP  bool // temporary opt-out of the new mapping logic
 	m        sync.RWMutex
 }
 
-func newMetadataMapperBundle() *MetadataMapperBundle {
-	return &MetadataMapperBundle{
-		Services: make(ServicesMapper),
+func newMetadataMapperBundle() *metadataMapperBundle {
```

Comment:
I think declaring `newMetadataMapperBundle` and `metadataMapperBundle` here when they are also in `types.go` https://github.com/DataDog/datadog-agent/pull/3145/files#diff-bb69630d823d5ad8bc550b836e716bf2R96 is confusing. It is not clear which one I should use depending on my use case.
If we have a types builder, all getters/setters and init should be there. Unless there is a reason I'm missing

Reply:
`apiserver.metadataMapperBundle` struct still exist because it need for have a mutex for concurrent access on its items. 
I will rename the struct in the `api/v1` to avoid confusion. (just need to get inspired :) )